fusedrug/data/tokenizer/ops/injector_tokenizer_ops.py

```diff
@@ -0,0 +1,177 @@
+from fuse.utils import NDict
+
+# from fuse.data import OpBase, get_sample_id
+from fusedrug.data.tokenizer.injectortokenizer.injector_tokenizer import (
+    InjectorTokenizer,
+)
+
+# from fusedrug.data.tokenizer.modulartokenizer.modular_tokenizer import ModularTokenizer
+from fusedrug.data.tokenizer.ops import FastModularTokenizer
+
+# from warnings import warn
+# from collections import defaultdict
+from typing import Optional, Union, Any
+
+# import os
+# import re
+# import torch
+
+
+class InjectorTokenizerOp(FastModularTokenizer):
+    """
+    applies a injector tokenizer
+
+    injector tokenizer builds on top of modular tokenizer.
+    its purpose is to build inputs_emb for the model (instead of input_ids)
+        this allows to support more advanced inputs beyond token ids, like:
+        * scalars inputs
+        * embeddings vector within a single input
+    """
+
+    def __init__(
+        self,
+        input_dim: int,
+        tokenizer_path: str,
+        max_size: Union[int, None] = None,
+        pad_token: Union[str, None] = None,
+        pad_type_id: Union[int, None] = None,
+        validate_ends_with_eos: Optional[bool] = True,
+        eos: Optional[str] = "<EOS>",
+        verbose: Optional[bool] = False,
+        **kwargs: Any,
+    ) -> None:
+        """
+
+        Args:
+            tokenizer_path: full path to a directory that the tokenizer will be loaded from
+            max_size: sequences below this size will be padded, and above this size will be truncated
+            pad: a string of the pad token
+            pad_type_id: see tokenizers.Tokenizer.enable_padding() docstring
+            validate_ends_with_eos: during encoder request (a _call_ to the op) will make sure that it ends with the provided eos token, and raise exception otherwise.
+                having an eos (end of sentence) token in the end is useful for multiple scenarios, for example in a generative transformer (like T5 encoder-decoder)
+            verbose:
+        """
+        if verbose:
+            print(
+                f"DEBUG:InjectorTokenizerOp __init__ called for path {tokenizer_path}"
+            )
+
+        super().__init__(
+            tokenizer_path=tokenizer_path,
+            max_size=max_size,
+            pad_token=pad_token,
+            pad_type_id=pad_type_id,
+            validate_ends_with_eos=validate_ends_with_eos,
+            eos=eos,
+            verbose=verbose,
+            **kwargs,
+        )
+
+        self._input_dim = input_dim
```

Comment:
are you using it somewhere? the functions below are static.

Reply:
removed it from here and also from multiple related places.